src/Platforms/Echo.Platforms.Dnlib/CilArchitecture.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using dnlib.DotNet;
+using dnlib.DotNet.Emit;
+using Echo.Core.Code;
+using IVariable = Echo.Core.Code.IVariable;
+
+namespace Echo.Platforms.Dnlib
+{
+    /// <summary>
+    /// Provides a description of the CIL instruction set architecture (ISA) that is modelled by dnlib.   
+    /// </summary>
+    public class CilArchitecture : IInstructionSetArchitecture<Instruction>
+    {
+        private readonly IList<CilVariable> _variables;
+        private readonly IList<CilParameter> _parameters;
+
+        /// <summary>
+        /// Creates a new CIL architecture description based on a CIL method body.
+        /// </summary>
+        /// <param name="method">The method definition.</param>
+        public CilArchitecture(MethodDef method)
+        {
+            Method = method ?? throw new ArgumentNullException(nameof(method));
+
+            if (!method.HasBody || method.Body is null)
+                throw new ArgumentException("Method does not have a CIL method body.", nameof(method));
+
+            _variables = MethodBody.Variables.Select(v => new CilVariable(v)).ToArray();
```

Comment:
To keep things consistent, put Linq method chains on separate lines. 

Refer to this:
https://github.com/Washi1337/Echo/blob/master/CONTRIBUTING.md#usage-of-linq